backend/lib/models/Notification.js

```diff
@@ -0,0 +1,58 @@
+const { Schema, model } = require("mongoose");
+
+const notificationSchema = new Schema(
+  {
+    action: {
+      enum: ["like", "comment", "post"],
```

Comment:
```suggestion
      enum: ["message", "like", "comment", "post"],
```

Reply:
@ddvloper messages notifications are not stored here, we can directly query the Threads collection with `newMessages: true`

Reply:
Does that mean Lambda will query Threads also for sending email notifications? I thought Socket will generate these notifications in the same format and Lambda then will query Notifications only

Reply:
We don't need to create a notification document for every message, all we need is to notify the user that they have X unread messages from Y. which is already stored on the Threads collection, so just using that makes more sense.

Reply:
Thanks @Zaydme